### DISCUSSION TITLE
Rendering other standard components with TinaMarkdown

### DISCUSSION THREAD

**Discussion by Figumari** [tinacms/tinacms#4102] [Help]
Hey everyone,

I've been exploring TinaCMS and its TinaMarkdown feature for managing rich-text content. It's been great so far, but I have a question. How can I render rich-text content using other standard components by using TinaMarkdown? For example, let's say I have some Tailwind components like `<h1 className="title-font text-3xl"/>` or `<p className="mb-8 text-base md:text-lg"/>`  or a next/link `<Link to"/"/>` – can I use them together with TinaMarkdown?

I'm looking for something like this:
```js
export default function RichText(props) {
  const {content} = props;

  const serializer = {
    heading1: ({ children }) => `<h1 className="title-font text-3xl">${children}</h1>`,
    paragraph: ({ children }) => `<p className="mb-8 text-base md:text-lg>${children}</p>`,
    hyperlink: ({ node, children }) => {
      const target = node.data.target
        ? `target="${node.data.target}" rel="noopener"`
        : ''
      const url = node.data
      return `<Link ${target} to="${url}">${children}</Link>`
    },
  };

  return (
    <TinaMarkdown serializer={serializer} content={content}/>
  );
}
```

Looking forward to hearing your thoughts and ideas on this! Thanks in advance for your help! 😊

**Comment by jeffsee55** [comment]
You can render any component with the `components` prop https://tina.io/docs/reference/types/rich-text/#using-object-object.

```tsx
import Image from 'next/image'

export default function RichText(props) {
  const {content} = props;

  return (
    <TinaMarkdown components={{img: props => <Image src={props.url} />}} content={content}/>
  );
}
```
I think `url` is the value, might need to double check that.
